src/main/scala/com/fulcrumgenomics/internal/FgMetricsDoclet.scala

```diff
@@ -0,0 +1,164 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.internal
+
+import java.io.PrintStream
+import java.nio.file.Paths
+
+import com.fulcrumgenomics.util.Metric
+
+import scala.tools.nsc.doc.base.comment._
+import scala.tools.nsc.doc.html.Doclet
+import scala.tools.nsc.doc.model.{DocTemplateEntity}
+
+/** Case class to capture information about a field/column in a metrics class/file. */
+case class ColumnDescription(name: String, typ: String, description: String)
+
+/** Case class to capture information about a metrics class/file. */
+case class MetricDescription(name: String, description: String, columns: Seq[ColumnDescription]) {
+  def summary: String = description.takeWhile(_ != '.').replace('\n', ' ')
+}
+
+/**
+  * Custom scaladoc Doclet for rendering the documentation for [[Metric]] classes into
+  * MarkDown for display on the fgbio website.
+  */
+class FgMetricsDoclet extends Doclet {
+  /**
+    * Main entry point for the doclet.  Scans for documentation for the metrics classes and
+    * renders it into MarkDown.
+    */
+  override def generateImpl(): Unit = {
+    // The MarkDown file to be written
+    val md  = Paths.get(this.universe.settings.outdir.value).resolve("metrics.md")
+    val out = new PrintStream(md.toFile)
+
+    out.println(
+      s"""
+         |# fgbio Metrics Descriptions
+         |
+         |This page contains descriptions of all metrics produced by all fgbio tools.  Within the descriptions
+         |the type of each field/column is given, including two commonly used types:
+         |
+         |* `Count` is a 64-bit integer representing the count of some item
+         |* `Proportion` is a 64-bit real number with a value between 0 and 1 representing a proportion or fraction
```

Comment:
I'll remove the 64-bit since that probably is too detailed for user docs.  I think more people will understand "integer" than "real number" though, so I'm going to leave that there.